src/hotspot/share/prims/jvmtiClassFileReconstituter.cpp

```diff
@@ -883,7 +883,7 @@ void JvmtiClassFileReconstituter::write_class_file_format() {
   copy_cpool_bytes(writeable_address(cpool_size()));
 
   // JVMSpec|           u2 access_flags;
-  write_u2(ik()->access_flags().get_flags() & JVM_RECOGNIZED_CLASS_MODIFIERS);
+  write_u2(ik()->access_flags().get_flags() & (JVM_RECOGNIZED_CLASS_MODIFIERS | JVM_ACC_INLINE));
```

Comment:
Would it better to consider adding JVM_ACC_INLINE to JVM_RECOGNIZED_CLASS_MODIFIERS?
Also, I see other definitions of the JVM_RECOGNIZED_CLASS_MODIFIERS in files:
```
 src/hotspot/share/include/jvm_constants.h
 src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/runtime/ClassConstants.java
```
Do we want this modifier added there as well?

Reply:
The problem is that JVM_RECOGNIZED_CLASS_MODIFIERS contains all modifiers defined for JDK8 and earlier, but the validity of new modifiers depends on class file version (see classFileParser.cpp where JVM_RECOGNIZED_CLASS_MODIFIERS is also used).

Reply:
Thanks, Frederic. Of course, it is always possible to add this modifier conditionally depending on the release. But it is not that elegant.